### ISSUE TITLE
Improve display of Grids

### ISSUE THREAD

**Issue by sarjon** [PrestaShop/PrestaShop#14018]
I feel like UI could be improved a little on grids. There is a lot of free space that pushes grid down. Also there is white space in header and this adds up to a lot of unused space on page in my opinion.

![download (1)](https://user-images.githubusercontent.com/15143151/58567036-4773db80-823a-11e9-91c9-26748e5c3c12.png)

Maybe "Select all" could be moved on the same line as "Bulk actions" button or maybe they could both be moved somewhere else or maybe it's not even an issue. :smiley: 

You can close this issue if you think that current UI is fine.

**Quickwin specs:**
The quickwin could be to display the button bulk action only **when one checkbox is selected.**

**Comment by khouloudbelguith** [comment]
ping @PrestaShop/prestashop-core-developers what do you think?

Thanks!

**Comment by Hlavtox** [comment]
I think Bulk Actions should be moved to the bottom. Because... what do you do? You select things you want to do things with, now you are on the bottom of the page, you will find bulk actions there.

Also, Select all could be done as a checkbox in the table header, next to ID.

**Comment by MatShir** [comment]
@TristanLDD @prestascott What do you think? It could be reworked with the new save button as well :thinking:

**Comment by Hlavtox** [comment]
@MatShir @prestascott 

![58567036-4773db80-823a-11e9-91c9-26748e5c3c12](https://user-images.githubusercontent.com/6097524/97164222-17f43600-1782-11eb-9d47-a0b83f351c9a.png)

**Comment by prestascott** [comment]
Hello everyone!

Thank you @sarjon for your report. :) 

I agree with you, there is a lot of space loss but what do you want to put on the first red box? Indeed, we can enlarge the search bar. 

'Select all' can be removed and put in the table as @Hlavtox said, and I think this is where the checkbox belongs to because it's now a standard for tables. 

I understand your point of view by placing the bulk actions button at the bottom, but what happens if you don't want to use it or just want to apply bulk actions on the first two rows that are selected? The button will be hidden if it's placed at the end of the page and will be difficult to reach. 

If you need to apply bulk actions after you selected a lof of things you want to do things with, maybe we can make the bulk actions button sticky at the top so it becomes reachable at any height of the page, so you won't have to scroll up back to the top.

**Comment by Hlavtox** [comment]
@prestascott Good point! What if we put it to the card/table header? There is a lot of free space. No proper icon comes to my mind though.

**Comment by prestascott** [comment]
Thanks!

I tried to combine the 'select all' with the bulk actions button and put it in the table header and it looks like this:
![Desktop - 2](https://user-images.githubusercontent.com/54888056/98958290-6df70680-2502-11eb-9c10-616300fb170b.png)
![Desktop - 3](https://user-images.githubusercontent.com/54888056/98958297-6f283380-2502-11eb-9f9e-96966c4f1b1b.png)

We are surely optimizing spaces and it's not that bad to be honest. 

But imo, i'd rather leave the bulk actions button at the top of the table because it's still more affordable than my design proposition. 

What we can do is to put the 'select all' in the table header and leave the bulk actions button where it is. 

Wdyt?

**Comment by hibatallahAouadni** [comment]
Hello @prestascott 

I definitely like your proposition :clap: It's a brilliant idea to combine these two components and delete that space:

![image](https://user-images.githubusercontent.com/23308427/104851749-2bcdb500-58f7-11eb-9ffe-91ffc3dd53e2.png)


Ping @PrestaShop/product-team  @sarjon @Hlavtox do you agree with me?

**Comment by Hlavtox** [comment]
Hi everyone!

@prestascott Sorry for not answering. On second thought, I agree to keep bulk actions on the top also, because when you want to do something with all items, you don't have to scroll down... I would try to find a place for it for top and bottom.

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Didn't you like the idea to combine the **select all** with the **bulk actions** button suggested by @prestascott 

![image](https://user-images.githubusercontent.com/23308427/104852567-07c0a280-58fc-11eb-8b1d-2f84b8afc02a.png)

Please check and feedback.
Thanks!

**Comment by Hlavtox** [comment]
@hibatallahAouadni @prestascott It's goood, but I think that unless you really know about it, you would never find it. It looks like a sort icon. But I can't think of a better solution, but there must be some... 😄

- I would keep the whole big bulk actions button on the bottom.
- To the top, we must think of some solution like @prestascott did, but can we make it somehow more readable? Space wise, its perfect!

**Comment by hibatallahAouadni** [comment]
@Hlavtox you're right :confused: how about this, check the screen record below:

[issue_14018.webm](https://github.com/PrestaShop/PrestaShop/assets/23308427/c1f767dc-9eec-4a54-9bd7-ef797eec8255)

Well, it's my humble proposition :sweat_smile: 
@prestascott I wish you like it :relaxed: (we may change the icon into somthing else if you would like :wink: )

**Comment by Hlavtox** [comment]
Yo @prestascott you got some competition right here 😄
@hibatallahAouadni That's better! What if you put a pencil or "three dot android menu" icon?

**Comment by MatShir** [comment]
Initially, the bulk action was moved up because it was not a practical position under the list. Mainly for having to scroll down to do the actions and it was never noticed by the beginners. I will rather not to move down again after the feedback we gather. So I guess @prestascott and @hibatallahAouadni have to fight for the first solution and have a nice integration on the top of the list.

**Comment by prestascott** [comment]
Hey everyone,

@hibatallahAouadni That's a really nice proposition!
I understand that the arrow next to the checkbox in my proposition can look like a filter/sort but I don't think merchants will be confused by it because it will only appear when boxes are checked. So it's pretty much understandable that it's about editing rows they've selected. I can redesign my proposition and try to make the checkbox looks like a button, so merchants will be never confused.

![Frame 2](https://user-images.githubusercontent.com/54888056/104903166-f3be8480-597f-11eb-9480-1d17333e527a.png)


Plus, this design is really common on tables. For example on [Google Mail](https://www.pinterest.fr/pin/43487952639718909/).

**Comment by Hlavtox** [comment]
@MatShir @prestascott What do you think about my idea of putting it both top and bottom? Because when you leave it only on the top, it would drive the other group crazy 😄.

And maybe you are right about the arrow, if Google does it. Just sometimes you design stuff like it's for blind people (nice descriptive button labels etc.), and this is like a PRO level interface 😄

**Comment by prestascott** [comment]
@Hlavtox It's a good idea to put both top and bottom but I don't think it's the best solution if we use my last design proposition. We can make the table header sticky so the bulk actions is accessible at any moment for merchants. Or, we can also add a button that bring merchants to the top of the table, not a fan but it's an alternative. :)

**Comment by hibatallahAouadni** [comment]
@Hlavtox I agree with @prestascott making the header sticky is the best solution :clap:

**Comment by ElodieOS** [comment]
Hi everyone! Is this issue still relevant? I like prescott's proposal and I think it's a good idea. 
I see a similar component has been created on the Product v2 page, so maybe it's a UX check?

**Comment by ElodieOS** [comment]
Hello @hibatallahAouadni and @Hlavtox 

Is this issue still relevant? I like prescott's proposal and I think it's a good idea to respond to this issue.

Do you need another proposition or it's an UX check?

**Comment by hibatallahAouadni** [comment]
Hello @ElodieOS 

And we like @prestascott proposal too :relaxed: We're waiting for @Hlavtox PR :rocket: 
Friendly reminder @Hlavtox :grin: 

Thanks!

**Comment by MatShir** [comment]
The quickwin could be to display the button bulk action only **when one checkbox is selected.**

**Comment by Hlavtox** [comment]
@MatShir I don't think so, you would think "where are the bulk actions". So you check some checkbox and wonder "What now?"... "oh some button appeared randomly on top and bottom.

Also, it would move the page after clicking, that's bad UX.

**Comment by MatShir** [comment]
yeah, you are probably right, but I feel @prestascott target more digital native. Just sharing my opinion

**Comment by boo-code** [comment]
The quickwin in the description is already in place on `9.2.x`, which is worth recording before someone implements it again.

`Blocks/bulk_actions.html.twig` renders the button with a hard-coded `disabled` attribute:

```twig
<button class="btn btn-outline-secondary dropdown-toggle js-bulk-actions-btn" data-toggle="dropdown" disabled>
```

and `bulk-action-checkbox-extension.ts` is what lifts it:

```ts
const checkedRowsCount = grid.getContainer().find(GridMap.bulks.checkedCheckbox).length;
if (checkedRowsCount > 0) { this.enableBulkActionsBtn(grid); } else { this.disableBulkActionsBtn(grid); }
```

So the button is inert until a row is checked and goes back to inert when the last one is unchecked. It is disabled rather than hidden, which is the only difference from the wording of the quickwin.

What the screenshot and the discussion above it are about - the vertical space above the grid, and whether "Select all" belongs on the same line as the button - is untouched by that and is still the open part.
